### PR TITLE
Issue 2304: docs for workaround

### PR DESCRIPTION
**Pull request by jenmalloy**

### CODE REVIEW

File: documentation/modules/ref-cpp-example.adoc [hunk 1]
```diff
@@ -73,6 +73,26 @@ int main(int argc, char **argv) {
...
 == Known issue with creating a subscriber on a hierarchical topic
 
-A known issue exists where creating a subscriber on a hierarchical topic in {ProductName} causes the broker to instead create it as a competing consumer (handling the address like a queue rather than a topic). The workaround involves setting the capability topic in the source. For more detailed information about how to add this capability, see link:https://github.com/ssorj/messaging-examples/tree/master/qpid-proton-cpp/auto-create[Add auto-create examples for {cpp}^].
+A known issue exists where creating a subscriber on a hierarchical topic in {ProductName} causes the broker to instead create it as a competing consumer (handling the address like a queue rather than a topic).
+
+The workaround involves setting the capability `"topic"` in the source.
+
+.Procedure
+
+. In the `topic_receive.cpp` file, edit the code so that it is similar to what is shown in this example:
```
Comment (ncbaratta) [inline documentation/modules/ref-cpp-example.adoc]: Should this be a numbered list if there is only 1 item?

File: documentation/modules/ref-python-example.adoc [hunk 2]
```diff
@@ -37,3 +37,28 @@ class HelloWorld(MessagingHandler):
...
+== Known issue with creating a subscriber on a hierarchical topic
+
+A known issue exists where creating a subscriber on a hierarchical topic in {ProductName} causes the broker to instead create it as a competing consumer (handling the address like a queue rather than a topic).
+
+The workaround for this issue involves setting the capability `"topic"` in the source.
+
+.Procedure
+
+. In the `simple_recv.py` file, add the following lines:
+
```
Comment (vbusch) [inline documentation/modules/ref-python-example.adoc]: Must also modify the `from proton.reactor import Container, ReceiverOption` to add the `ReceiverOption`

File: documentation/modules/ref-python-example.adoc [hunk 3]
```diff
@@ -37,3 +37,28 @@ class HelloWorld(MessagingHandler):
...
+    def apply(self, receiver):
+        receiver.source.capabilities.put_object(symbol("topic"))
+----
+
+. Modify the following line to add `options=CapabilityOptions()`:
+
+[source,python,options="nowrap",subs="+quotes,attributes"]
+----
+def on_start(self, event):
+        event.container.create_receiver(self.url, options=CapabilityOptions())
```
Comment (vbusch) [inline documentation/modules/ref-python-example.adoc]: There are 2 similar python examples in our documentation:
- common/client-example1.py (same as: contributing/common/client-example1.py, service_admin/common/client-example1.py, tenant/common/client-example1.py, address_space_admin/common/client-example1.py, messaging_app_developer/common/client-example1.py
- modules/ref-python-example.adoc

For this file (modules/ref-python-example.adoc) the line needs to be: 
`event.container.create_receiver(conn, self.address, options=CapabilityOptions())`
For all other locations `event.container.create_receiver(self.url, options=CapabilityOptions())`

File: documentation/modules/con-standard-topic.adoc [hunk 4]
```diff
@@ -27,3 +27,7 @@ So, for example:
 
 In the standard address space, the first level must always be a defined topic address; that is, `#` and `+` are not valid as the first characters of a subscribing address.
 
+== Known issue with creating a subscriber on a hierarchical topic
+
+A known issue exists where creating a subscriber on a hierarchical topic in {ProductName} causes the broker to instead create it as a competing consumer (handling the address like a queue rather than a topic). For more information about the specific workaround for your client, see the applicable client example section in link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#assembly-connecting-applications-messaging[].
```
Comment (ncbaratta) [inline documentation/modules/con-standard-topic.adoc]: For a link you need to fill in text in the [] .. it's only xrefs that fill those in automatically.

File: documentation/modules/ref-python-example.adoc [hunk 5]
```diff
@@ -37,3 +37,28 @@ class HelloWorld(MessagingHandler):
...
 
+== Known issue with creating a subscriber on a hierarchical topic
+
+A known issue exists where creating a subscriber on a hierarchical topic in {ProductName} causes the broker to instead create it as a competing consumer (handling the address like a queue rather than a topic).
+
+The workaround for this issue involves setting the capability `"topic"` in the source.
+
+.Procedure
+
+. In the `simple_recv.py` file, modify the `from proton.reactor import Container` to add the `ReceiverOption`:
```
Comment (ncbaratta) [inline documentation/modules/ref-python-example.adoc]: Shouldn't it be a bullet if it's only 1 item?

File: documentation/modules/con-standard-topic.adoc (hunk 4 above)
Comment (jenmalloy) [reply documentation/modules/con-standard-topic.adoc]: Done

File: documentation/modules/ref-python-example.adoc (hunk 5 above)
Comment (jenmalloy) [reply documentation/modules/ref-python-example.adoc]: Done

File: documentation/modules/ref-python-example.adoc (hunk 3 above)
Comment (jenmalloy) [reply documentation/modules/ref-python-example.adoc]: Fixed

File: documentation/modules/ref-python-example.adoc (hunk 2 above)
Comment (jenmalloy) [reply documentation/modules/ref-python-example.adoc]: Fixed

File: documentation/modules/con-brokered-topic.adoc [hunk 6]
```diff
@@ -24,4 +24,7 @@ So, for example:
 
 * `a/+/b` matches `a/foo/b` and `a/bar/b`, but would not match `a/foo/bar`
 
+== Known issue with creating hierarchical topics and clients
+
+A known issue exists where creating a multicast topic in {ProductName} causes the {BrokerName} to instead create it as an anycast topic. For more information about the specific workaround for your client, see the applicable client example section in link:{BookUrlBase}{BaseProductVersion}{BookNameUrl}#assembly-connecting-applications-messaging[].
```
[the remark this answers is not in the document]
Comment (jenmalloy) [reply documentation/modules/con-brokered-topic.adoc]: Fixed

**Comment by jenmalloy** [comment]
@rgodfrey Can you re-review please? Thanks!

**Pull request merged**